public/app/features/explore/ExplorePage.test.tsx

```diff
@@ -0,0 +1,65 @@
+import { fireEvent, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import { setupExplore } from './spec/helper/setup';
+
+const fetch = jest.fn().mockResolvedValue({ correlations: [] });
+jest.mock('@grafana/runtime', () => ({
+  ...jest.requireActual('@grafana/runtime'),
+  getBackendSrv: () => ({ fetch }),
+}));
+
+jest.mock('rxjs', () => ({
+  ...jest.requireActual('rxjs'),
+  lastValueFrom: () =>
+    new Promise((resolve, reject) => {
+      resolve({ data: { correlations: [] } });
+    }),
+}));
+
+function mockBoundingRectWidth(element: Element, width: number) {
+  element.getBoundingClientRect = (): DOMRect => {
+    return {
+      x: 0,
+      y: 0,
+      toJSON: () => '',
+      bottom: 0,
+      height: 0,
+      left: 0,
+      right: 0,
+      top: 0,
+      width,
+    };
+  };
+}
+
+describe('ExplorePage', () => {
+  it('updates split size when dragging', async () => {
```

Comment:
i think this test is not really testing anything in our code but rather a combination of how the split library we use works and what the browser should return, which imho is not adding any value.

there's some untested logic in ExplorePage that we should probably test insted by assuming this works correctly:

```tsconst updateSplitSize = (size: number) => {
    const evenSplitWidth = windowWidth / 2;
    const areBothSimilar = inRange(size, evenSplitWidth - 100, evenSplitWidth + 100);
    if (areBothSimilar) {
      dispatch(splitSizeUpdateAction({ largerExploreId: undefined }));
    } else {
      dispatch(
        splitSizeUpdateAction({
          largerExploreId: size > evenSplitWidth ? panes[1][0] : panes[0][0],
        })
      );
    }

    setRightPaneWidthRatio(size / windowWidth);
  };

  let widthCalc = 0;
  if (hasSplit) {
    if (!exploreState.evenSplitPanes && exploreState.maxedExploreId) {
      widthCalc = exploreState.maxedExploreId === panes[1][0] ? windowWidth - minWidth : minWidth;
    } else if (exploreState.evenSplitPanes) {
      widthCalc = Math.floor(windowWidth / 2);
    } else if (rightPaneWidthRatio !== undefined) {
      widthCalc = windowWidth * rightPaneWidthRatio;
    }
  }
```

i think by moving this logic into an hook and by testing that hook instead (basically rerendering it with arbitrary values) we can check that the logic is correct, the correct actions are dispatched and `widthCalc` is calculated properly. then `SplitPaneWrapper` from this component POV is really just an external, nothing that we should really test (as in it's guaranteed that moving the resizer triggers that action, we are only responsible for what comes after)

Reply:
Thanks for the feedback! I've updated this.